src/zebra/payment_methods.clj

```diff
@@ -5,19 +5,20 @@
            [com.stripe.net RequestOptions]
            [java.util Map]))
 
-(defn payment-method->map [x]
-  (merge
-    {:id     (.getId x)
-     :object (.getObject x)}
-    (when-let [card (.getCard x)]
-      {:card {:brand     (.getBrand card)
-              :exp_month (.getExpMonth card)
-              :exp_year  (.getExpYear card)
-              :funding   (.getFunding card)
-              :last4     (.getLast4 card)
-              :three_d_secure_usage
-                         {:supported
-                          (-> card .getThreeDSecureUsage .getSupported)}}})))
+(defn payment-method->map [^PaymentMethod x]
+  (with-meta (merge
+               {:id     (.getId x)
+                :object (.getObject x)}
+               (when-let [card (.getCard x)]
+                 {:card {:brand     (.getBrand card)
+                         :exp_month (.getExpMonth card)
+                         :exp_year  (.getExpYear card)
+                         :funding   (.getFunding card)
+                         :last4     (.getLast4 card)
+                         :three_d_secure_usage
+                                    {:supported
+                                     (-> card .getThreeDSecureUsage .getSupported)}}}))
+             {:original x}))
```

Comment:
opted for a generic name here in case the pattern looks good and could be applied in other places.

Reply:
Makes sense